pkg/controllers/migration/virtualmachine.go

```diff
@@ -214,10 +230,23 @@ func (h *virtualMachineHandler) triggerExport(vm *migration.VirtualMachineImport
 
 	if util.ConditionExists(vm.Status.ImportConditions, migration.VirtualMachinePoweredOff, v1.ConditionTrue) &&
 		util.ConditionExists(vm.Status.ImportConditions, migration.VirtualMachinePoweringOff, v1.ConditionTrue) &&
-		!util.ConditionExists(vm.Status.ImportConditions, migration.VirtualMachineExported, v1.ConditionTrue) {
+		!util.ConditionExists(vm.Status.ImportConditions, migration.VirtualMachineExported, v1.ConditionTrue) &&
+		!util.ConditionExists(vm.Status.ImportConditions, migration.VirtualMachineExportFailed, v1.ConditionTrue) {
```

Comment:
NIT, just afraid that `if` condition becomes very long at one day.

```go
type ConditionCheck struct {
	Type   condition.Cond
	Status v1.ConditionStatus
	NegateLogic bool
}

func ConditionsExists(conditions []common.Condition, checks []ConditionCheck) bool {
	for _, check := range checks {
		exists := ConditionExists(conditions, check.Type, check.Status)
		if check.NegateLogic {
			if exists {
				return false
			}
		} else {
			if !exists {
				return false
			}
		}
	}
	return true
}

func ConditionExists(conditions []common.Condition, c condition.Cond, condType v1.ConditionStatus) bool {
	for _, v := range conditions {
		if v.Type == c && v.Status == condType {
			return true
		}
	}
	return false
}

checks := []ConditionCheck{
	{Type: migration.VirtualMachinePoweredOff, Status: v1.ConditionTrue, NegateLogic: false},
	{Type: migration.VirtualMachinePoweringOff, Status: v1.ConditionTrue, NegateLogic: false},
	{Type: migration.VirtualMachineExported, Status: v1.ConditionTrue, NegateLogic: true},
	{Type: migration.VirtualMachineExportFailed, Status: v1.ConditionTrue, NegateLogic: true},
}

if util.ConditionsExists(vm.Status.ImportConditions, checks) {
	// Your code here
}
```

Reply:
i agree, but for now i find it easier to read conditions. may be in future changes we can rework this logic better.